2996_reflection/reflection.md

```diff
@@ -633,15 +633,52 @@ constexpr auto struct_to_tuple(From const& from) {
     args.push_back(reflect_value(mem));
   }
 
-  auto f = entity_ref<To(From const&)>(substitute(^struct_to_tuple_helper, args));
+  auto f = value_of<To(&)(From const&)>(substitute(^struct_to_tuple_helper, args));
   return f(from);
 }
 ```
 :::
 
-Here, `struct_to_tuple_type` takes a reflection of a type like `struct { T t; U const& u; V v; }` and returns a reflection of the type `std::tuple<T, U, V>`. `helper`. That gives us the return type. Then, `struct_to_tuple_helper` is a function template that does the actual conversion - which it can do by having all the reflections of the members as a non-type template parameter pack.
+Here, `struct_to_tuple_type` takes a reflection of a type like `struct { T t; U const& u; V v; }` and returns a reflection of the type `std::tuple<T, U, V>`.
+That gives us the return type.
+Then, `struct_to_tuple_helper` is a function template that does the actual conversion --- which it can do by having all the reflections of the members as a non-type template parameter pack.
 
-Everything is put together by using `substitute` to create the instantiation of `struct_to_tuple_helper` that we need, which is use `entity_ref` to get the correct function out of. `f` there is a function pointer to the correct specialization of `struct_to_tuple_helper`. Which we can simply invoke.
+Everything is put together by using `substitute` to create the instantiation of `struct_to_tuple_helper` that we need, and a compile-time reference to that instance is obtained with `value_of`.
+Thus `f` is a function reference to the correct specialization of `struct_to_tuple_helper`, which we can simply invoke.
+
+## Compile-Time Ticket Counter
+
+The features proposed here make it a little easier to update a ticket counter at compile time.
+This is not an ideal implementation (we'd prefer direct support for compile-time —-- i.e., `consteval` --- variables), but it shows how compile-time mutable state surfaces in new ways.
+
+::: bq
+```cpp
+class TU_Ticket {
+  template<int N> struct Helper {
+    constexpr int value = N;
```

Comment:
I think this needs to be:
```suggestion
    static constexpr int value = N;
```
And then, would `value_of<int>(static_data_members_of(r)[0])` actually work? We don't currently have a description of `value_of` for a static data member.

Reply:
We say:
> If r is a reflection for a constant-expression or a constant-valued entity of type T, value_of<T>(r) evaluates to that constant value.

that covers constexpr static data members, IMO.  But we do need to add `static_data_members_of` then.